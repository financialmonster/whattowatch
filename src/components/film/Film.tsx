import React, { FC, memo } from 'react';
import { Map } from 'immutable';

import { useFilmsPagePush } from 'hooks/useFilmsPagePush';

type TFilmProps = {
    film: Map<string, any>;
}

export const Film: FC<TFilmProps> = memo(({ film }) => {
    const name: string = film.get(`name`);
    const id: number = film.get(`id`);

    const filmDivClickHandler = useFilmsPagePush(id);

    return (
        <article className="small-movie-card catalog__movies-card" title={name}>
            <div className="small-movie-card__image" onClick={filmDivClickHandler}>
                <img src={film.get(`preview_image`)} alt={name} width="280" height="175" />
            </div>
            <h3 className="small-movie-card__title">
                <a className="small-movie-card__link">
                    {name}
                </a>
            </h3>
        </article>
    );
});
