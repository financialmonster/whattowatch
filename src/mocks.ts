import { Map, fromJS, List } from 'immutable';

export const createStoreMock = (mode: string = `common`) => {
    return {
        promo: Map({
            promo: Map(),
            isPromoFetching: true,
            promoError: null
        }),
        films: Map({
            films: fromJS([{genre: `Thriller`, id: 1}, {genre: `Comedy`, id: 2}]),
            isFilmsFetching: mode === `detailed`,
            filmsError: null,
            filter: `All genres`
        }),
        auth: Map({
            user: null,
            authError: null,
            isAuthFetching: false,
            authStatusError: null,
            isAuthStatusFetching: false
        }),
        reviews: Map({
            reviews: List(),
            isReviewsFetching: true,
            reviewsError: null
        })
    }
}
