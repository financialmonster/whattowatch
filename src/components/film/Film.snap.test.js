import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { Film } from './Film';
import { createStoreMock, mockedFilm } from 'mocks';

it(`Film: correctly rendered`, () => {
    const mockStore = configureStore();
    const store = mockStore(createStoreMock());

    const tree = renderer.create(
        <Provider store={store}>
            <Film film={mockedFilm} />
        </Provider> 
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
