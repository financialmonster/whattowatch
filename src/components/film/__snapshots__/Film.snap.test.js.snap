// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film: correctly rendered 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <img
      height="175"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
    />
  </h3>
</article>
`;
